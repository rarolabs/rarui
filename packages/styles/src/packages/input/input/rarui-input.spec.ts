import { input } from ".";

describe("Input Styles", () => {
  it("should apply base container styles correctly", () => {
    const className = input.classnames.container();
    expect(className).toContain("container");
  });

  it("should apply appearance variants correctly", () => {
    const appearances = ["success", "error"] as const;
    appearances.forEach((appearance) => {
      const className = input.classnames.container({ appearance });
      expect(className).toContain(appearance);
      expect(className).toMatchSnapshot(`appearance-${appearance}`);
    });
  });

  it("should apply border variants correctly", () => {
    const borders = [true, false];
    borders.forEach((border) => {
      const className = input.classnames.container({ border });
      expect(className).toContain(border ? "true" : "false");
      expect(className).toMatchSnapshot(`border-${border}`);
    });
  });

  it("should apply size variants correctly", () => {
    const sizes = ["large", "medium", "small"] as const;
    sizes.forEach((size) => {
      const className = input.classnames.container({ size });
      expect(className).toContain(size);
      expect(className).toMatchSnapshot(`size-${size}`);
    });
  });

  it("should apply base input styles correctly", () => {
    const className = input.classnames.input();
    expect(className).toContain("input");
  });

  it("should apply input size variants correctly", () => {
    const sizes = ["large", "medium", "small"] as const;
    sizes.forEach((size) => {
      const className = input.classnames.input({ size });
      expect(className).toContain(size);
      expect(className).toMatchSnapshot(`input-size-${size}`);
    });
  });

  it("should apply leading styles correctly", () => {
    const className = input.classnames.leading();
    expect(className).toContain("leading");
  });

  it("should apply leading appearance variants correctly", () => {
    const appearances = ["success", "error"] as const;
    appearances.forEach((appearance) => {
      const className = input.classnames.leading({ appearance });
      expect(className).toContain(appearance);
      expect(className).toMatchSnapshot(`leading-appearance-${appearance}`);
    });
  });

  it("should apply leading position variants correctly", () => {
    const positions = ["start", "end"] as const;
    positions.forEach((position) => {
      const className = input.classnames.leading({ position });
      expect(className).toContain(position);
      expect(className).toMatchSnapshot(`leading-position-${position}`);
    });
  });
});
