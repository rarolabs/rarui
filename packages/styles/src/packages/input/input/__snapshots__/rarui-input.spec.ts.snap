// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Styles should apply appearance variants correctly: appearance-error 1`] = `"rarui-input_container__lq13ap0 rarui-input_container_appearance_error__lq13ap2"`;

exports[`Input Styles should apply appearance variants correctly: appearance-success 1`] = `"rarui-input_container__lq13ap0 rarui-input_container_appearance_success__lq13ap1"`;

exports[`Input Styles should apply border variants correctly: border-false 1`] = `"rarui-input_container__lq13ap0 rarui-input_container_border_false__lq13ap4"`;

exports[`Input Styles should apply border variants correctly: border-true 1`] = `"rarui-input_container__lq13ap0 rarui-input_container_border_true__lq13ap3"`;

exports[`Input Styles should apply input size variants correctly: input-size-large 1`] = `"rarui-input__lq13ap8 rarui-input_size_large__lq13apa"`;

exports[`Input Styles should apply input size variants correctly: input-size-medium 1`] = `"rarui-input__lq13ap8 rarui-input_size_medium__lq13apb"`;

exports[`Input Styles should apply input size variants correctly: input-size-small 1`] = `"rarui-input__lq13ap8 rarui-input_size_small__lq13apc"`;

exports[`Input Styles should apply leading appearance variants correctly: leading-appearance-error 1`] = `"rarui-input_leading__lq13apd rarui-input_leading_appearance_error__lq13apj"`;

exports[`Input Styles should apply leading appearance variants correctly: leading-appearance-success 1`] = `"rarui-input_leading__lq13apd rarui-input_leading_appearance_success__lq13api"`;

exports[`Input Styles should apply leading position variants correctly: leading-position-end 1`] = `"rarui-input_leading__lq13apd rarui-input_leading_position_end__lq13apf"`;

exports[`Input Styles should apply leading position variants correctly: leading-position-start 1`] = `"rarui-input_leading__lq13apd rarui-input_leading_position_start__lq13ape"`;

exports[`Input Styles should apply size variants correctly: size-large 1`] = `"rarui-input_container__lq13ap0 rarui-input_container_size_large__lq13ap5"`;

exports[`Input Styles should apply size variants correctly: size-medium 1`] = `"rarui-input_container__lq13ap0 rarui-input_container_size_medium__lq13ap6"`;

exports[`Input Styles should apply size variants correctly: size-small 1`] = `"rarui-input_container__lq13ap0 rarui-input_container_size_small__lq13ap7"`;
